internal/pkg/cli/app_init.go

```diff
@@ -161,7 +160,21 @@ func (o *initAppOpts) Execute() error {
 }
 
 func (o *initAppOpts) createManifest() (string, error) {
-	manifest, err := manifest.CreateApp(o.AppName, o.AppType, o.DockerfilePath)
+	props := &manifest.CreateAppManifestProps{
+		AppName:    o.AppName,
+		Dockerfile: o.DockerfilePath,
+	}
+	proj, err := o.projGetter.GetProject(o.ProjectName())
+	if err != nil {
+		return "", fmt.Errorf("get project %s: %w", o.ProjectName(), err)
+	}
+	o.proj = proj
```

Comment:
What do you think about fetching the project in the constructor instead?

I'm asking because setting `o.proj` in `createManifest` is not the most obvious place where you'd expense a member variable to be set.

Reply:
Yeah I agree. But if we move the fetching to the constructor then if a user is not in a project dir, which means the `o.ProjectName()` will be an empty string. Then, it should return `errNoProjectInWorkspace` instead of `fmt.Errorf("get project %s: %w")`, which means that we would probably need to move the project name validation into the constructor as well.

Do you think it is better to move the fetching to `Validate()` instead?

Reply:
oh i see then maybe at the beginning of `Execute` to make it explicit instead of hiding it in `createManifest`?

`Validate` should not be to set any hidden values. Another command should be able to set `Vars`, embed it into `Opts`, and call `Opts.Execute` without having to call `Validate` or `Ask`.